app/components/createService/createservice.html

```diff
@@ -445,6 +446,52 @@
           <!-- tab-placement -->
           <div class="tab-pane" id="placement" ng-include="'app/components/createService/includes/placement.html'"></div>
           <!-- !tab-placement -->
+          <!-- tab-resources -->
+          <div class="tab-pane" id="resources">
+            <form class="form-horizontal" style="margin-top: 15px;">
+              <div class="form-group">
+                <!-- cpu-limit-input -->
+                <label for="cpu-limit" class="col-sm-2 col-lg-2 control-label text-left">CPU limit<portainer-tooltip position="top" message="Maximum number of CPU the service can use (0 to unlimited)."></portainer-tooltip></label>
+                <div class="col-sm-4 col-lg-4">
+                  <input type="number" step="0.1" class="form-control" ng-model="formValues.CpuLimit" id="cpu-limit" placeholder="e.g. 0.5">
+                </div>
+                <!-- !cpu-limit-input -->
+                <!-- cpu-reservation-input -->
+                <label for="cpu-reservation" class="col-sm-2 col-lg-2 control-label text-left">CPU res<portainer-tooltip position="top" message="Minimum number of CPU guaranteed to the service."></portainer-tooltip></label>
+                <div class="col-sm-4 col-lg-4">
+                  <input type="number" step="0.1" class="form-control" ng-model="formValues.CpuReservation" id="cpu-limit" placeholder="e.g. 0.3">
+                </div>
+                <!-- !cpu-reservation-input -->
+              </div>
+              <div class="form-group">
+                <!-- memory-limit-input -->
+                <label for="memory-limit" class="col-sm-2 col-lg-2 control-label text-left">Mem limit<portainer-tooltip position="top" message="Maximum memory the service can use (0 to unlimited)."></portainer-tooltip></label>
+                <div class="col-sm-2 col-lg-2">
```

Comment:
What about using `col-sm-3` only here ? And `col-sm-1` only for the unit select ? Visually, the select is too big just for MB/GB.

**Sidenote**: when you write `class="col-sm-2 col-lg-2"` there is no need to specify the `col-lg-2` as it's the same value than `col-sm-2`, it will fallback automatically to `col-sm-2`. I've seen that in multiple places, you can just specify the `col-sm-*` attribute.

Reply:
With select in col-sm-1, display is truncated or not present, see here : 
Memory res is "2-2"
Memory limit is "3-1"
![image](https://user-images.githubusercontent.com/15888093/30504446-083edcda-9a6f-11e7-8680-f3a5e78820f3.png)

Reply:
Right, then we'll keep `col-sm-2` & `col-sm-2`